pkg/store/storepb/custom.go

```diff
@@ -257,6 +258,28 @@ func (s *uniqueSeriesSet) Next() bool {
 	return true
 }
 
+func (m AggrChunk) Hash() string {
```

Comment:
Seems the hash is a little bit expensive to calculate...

Reply:
Previously we were [comparing each chunk byte-by-byte using recursion and deduplicating this way.](https://github.com/thanos-io/thanos/blob/79e70da702228ac0282fc7639f9f160d922b6dcb/pkg/store/storepb/custom.go#L160-L203) So, in this regard the complexity is the same AIUI - we are _still_ reading each and every byte. Just that instead of comparison, we are now calculating hashes and then using them in a map.

I was thinking that maybe we could use md5 instead here as we don't care about collisions here really? In the very unlikely case of collisions, the upper layer will take care of deduplicating. MD5 is much faster. [We could use this](https://github.com/minio/md5-simd).